libtextsecure/helpers.js

```diff
@@ -90,6 +90,11 @@ window.textsecure.utils = function() {
         return number.split(".");
     };
 
+    self.isNumberSane = function(number) {
```

Comment:
Really, the point of this is so that I dont have to audit everything for "what if someone sends me a group update with a numebr of "+1XXXXXXXXXX.1" ie a device identifier. Its a sanity check, not a number-valid check, so even for eg. emails, we still need to be able to check like "is someone actively attacking us with unicode control chars?"